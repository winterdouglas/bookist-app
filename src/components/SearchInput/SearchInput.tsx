import {
  StyleProp,
  StyleSheet,
  TextInput,
  TextInputProps,
  TextStyle,
  View,
  ViewStyle,
} from "react-native";
import { Icon } from "@/components/Icon";
import { spacing } from "@/theme";
import { useAppTheme } from "@/hooks/useAppTheme";
import { useTranslation } from "react-i18next";

export type SearchInputProps = TextInputProps & {
  style?: StyleProp<ViewStyle>;
  inputStyle?: StyleProp<TextStyle>;
};

export const SearchInput = ({
  placeholder: inputPlaceholder,
  style,
  inputStyle,
  ...props
}: SearchInputProps) => {
  const { colors } = useAppTheme();
  const { t } = useTranslation();
  const placeholder = inputPlaceholder ?? t("search");

  const $containerStyles: StyleProp<ViewStyle> = [
    styles.container,
    { backgroundColor: colors.border },
    style,
  ];

  return (
    <View style={$containerStyles}>
      <TextInput
        underlineColorAndroid="transparent"
        placeholderTextColor={colors.textDim}
        {...props}
        placeholder={placeholder}
        style={[styles.input, inputStyle, { color: colors.text }]}
      />
      <Icon
        style={{ color: colors.textDim }}
        name="search"
        size={spacing.extraLarge}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    paddingVertical: spacing.tiny,
    paddingHorizontal: spacing.extraSmall,
    borderRadius: 100,
  },
  input: {
    padding: spacing.extraSmall,
    flex: 1,
  },
});
