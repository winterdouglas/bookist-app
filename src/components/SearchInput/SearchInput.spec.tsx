import renderer from "react-test-renderer";
import { SearchInput } from "./SearchInput";
import { ThemeProvider } from "@/contexts";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ThemeProvider>
        <SearchInput />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
