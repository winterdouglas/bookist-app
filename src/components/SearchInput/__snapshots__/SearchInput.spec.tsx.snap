// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 100,
        "flexDirection": "row",
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "rgb(216, 216, 216)",
      },
      undefined,
    ]
  }
>
  <TextInput
    placeholder="search"
    placeholderTextColor="#666666"
    style={
      [
        {
          "flex": 1,
          "padding": 8,
        },
        undefined,
        {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        {
          "color": "#666666",
        },
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    î¹ 
  </Text>
</View>
`;
